src/meadowrun/run_job.py

```diff
@@ -804,6 +899,81 @@ async def run_map(
     )
 
 
+async def run_map_as_completed(
+    function: Callable[[_T], _U],
+    args: Sequence[_T],
+    host: Host,
+    resources_per_task: Optional[Resources] = None,
+    deployment: Union[Deployment, Awaitable[Deployment], None] = None,
+    num_concurrent_tasks: Optional[int] = None,
+    sidecar_containers: Union[
+        Iterable[ContainerInterpreterBase], ContainerInterpreterBase, None
+    ] = None,
+    ports: Union[Iterable[str], str, Iterable[int], int, None] = None,
+) -> AsyncIterable[TaskResult[_U]]:
+    """
+    Equivalent to [run_map][meadowrun.run_map], but returns results from tasks as they
+    are completed.
+    Returns:
```

Comment:
Please add to the generated documentation (docs/reference/apis.md) and make sure the generated documentation looks good `poetry run mkdocs serve`